### ISSUE TITLE
Gray Bars on Sony AF8 after being updated to version 21. 

### ISSUE THREAD

**Issue by Kristijan1001** [xbmc/xbmc#25166]
## Bug report
Gray Bars on Sony TV after v21
### Describe the bug
Here is a clear and concise description of what the problem is:
Playing videos shows Gray Bars and Washed out UI instead of Black Bars. Enabling Dolby fixes it but video lags hard.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
Movie to have black bars.

## Actual Behavior
Movie has Gray Bars.

## Possible Fix
Apparently this was already fixed long time ago but for some reason reverted to fix some issues other people had. The thing is you can't just remove something that has been working for 5 Years. You just made the reverse now, we are having issues and this "other" people don't have them. How about a middle ground. Don't push fixes until both groups are satisfied or an option is introduced to enable the fix. 

![image](https://github.com/xbmc/xbmc/assets/24441639/2e797d79-b780-40f5-9536-a152c12b79ca)

https://forum.kodi.tv/showthread.php?tid=376806&pid=3189708

https://forum.kodi.tv/showthread.php?tid=371557&pid=3189636#pid3189636

https://github.com/xbmc/xbmc/pull/22561

### To Reproduce
1. Play a HDR/Dolby Movie
2. Look at Gray Bars

### Debuglog

The debuglog can be found here:
Not doing that sorry, Issue is well known.

### Screenshots 
Here are some links or screenshots to help explain the problem:
Gray vs Black

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
https://forum.kodi.tv/showthread.php?tid=376806&pid=3189708

https://forum.kodi.tv/showthread.php?tid=371557&pid=3189636#pid3189636

### Your Environment
Used Operating system:

 - [X ] Android

 - Operating system version/name:
 - Kodi version: 21

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by fritsch** [comment]
The "some reason" is that this Sony workaround broke Dolby Visions on multiple other devices ... good enough for "some reason" I'd say ...

**Comment by Kristijan1001** [comment]
> The "some reason" is that this Sony workaround broke Dolby Visions on multiple other devices ... good enough for "some reason" I'd say ...

We need a way to enable it back, it kinda sucks it worked for 5 years, now all of a sudden it doesn't. Maybe add an advancedsetting so people with the Hardware can enable it. You don't really fix an issue if you remove something that hasnt caused people issues for the last 5 years.

People can live without Dolby vision and stick with normal HDR until a middle ground is found. But those with SONY TV is basically unusable can't watch any HDR Stuff...

**Comment by fritsch** [comment]
Middle ground as in: Ask Sony to fix it? Your device was very expensive, right?
Or middle ground as in: use v20 as Dolby Vision is of no interest for you?

**Comment by fritsch** [comment]
For the record:
https://github.com/jellyfin/jellyfin-androidtv/issues/2967
https://ap.community.sony.com/s/question/0D54O00006o7JY2SAM/grey-bar-randomly-showing-up-at-the-bottom-of-the-screen-xbr43x800d?language=en_US
https://github.com/google/ExoPlayer/issues/8803
https://emby.media/community/index.php?/topic/57838-hdr-grey-bars/

And also for the record: Painting it "black" caused issues for everyone else.

As can be seen from above links: it's a Sony firmware bug, the vendor does not care.

**Comment by Kristijan1001** [comment]
> For the record: [jellyfin/jellyfin-androidtv#2967](https://github.com/jellyfin/jellyfin-androidtv/issues/2967) https://ap.community.sony.com/s/question/0D54O00006o7JY2SAM/grey-bar-randomly-showing-up-at-the-bottom-of-the-screen-xbr43x800d?language=en_US [google/ExoPlayer#8803](https://github.com/google/ExoPlayer/issues/8803) https://emby.media/community/index.php?/topic/57838-hdr-grey-bars/
> 
> And also for the record: Painting it "black" caused issues for everyone else.
> 
> As can be seen from above links: it's a Sony firmware bug, the vendor does not care.

Yes i know of this, and i know it is Sony fault, yes the TV is also expensive 2k $ 5 years ago. And Yes Sony doesnt care and i dont think they will ever fix it. They have moved on. 

Is there no way to make this toggable through a config no hope for the future  ?

**Comment by fritsch** [comment]
I will look at it tomorrow and see. I made the original fix. The transparency we basically added there cause even a performance issue for some machines. After all the FireTV devices started to appear - the workaround even caused active issues with the performance.

Fun-Fact: It even killed some LiveTV experience for other Sony users who need to watch LiveTV with SW decoder + deinterlacer - but I cannot remember all details - as you already said: it's 6 years ago.

Will check once more, if we can "actively" later the View's background color and make it dynamic. While I check again - I remember that this exactly was the issue. It's either one or the other.

@quietvoid I think you had a look at this as well, we basically could not change the color?

@Kristijan1001 there is one new setting for "OSD brightness" when playing HDR content - it's an overlay menu which tunes the shaders used for GLES output. Does changing this bar change the grey color?

**Comment by fritsch** [comment]
Okay - I read it up: https://github.com/xbmc/libandroidjni/pull/43 <- theoretically based on this, we could create such an advancedsetting.

@quietvoid what was your intention back then? Expose a setting?

**Comment by quietvoid** [comment]
> Okay - I read it up: https://github.com/xbmc/libandroidjni/pull/43 <- theoretically based on this, we could create such an advancedsetting.
> 
> @quietvoid what was your intention back then? Expose a setting?

It was an attempt at setting the background at runtime and it worked.
It should be usable for a setting.

**Comment by Kristijan1001** [comment]
> @Kristijan1001 there is one new setting for "OSD brightness" when playing HDR content - it's an overlay menu which tunes the shaders used for GLES output. Does changing this bar change the grey color?
I will have to try that and report back. Also just to mention again , its not only the black bars that are gray, the whole menu is like the whole OSD pause play buttons etc. It looks like HDR is being applied to the movie but not to the UI.

**Comment by Kristijan1001** [comment]
> Yes, that is what that new setting is for - so please report back if that makes a difference / changes the color tones - screenshots are welcome.

I tested it this morning before work i didnt see difference between 0% and 100%. When i get back i will test it again and take photos and compare to make sure.

**Comment by fritsch** [comment]
First testbuild: https://jenkins.kodi.tv/job/android-arm-docker/6548/artifact/kodi-20240509-62ad1ec2-decorview-armeabi-v7a.apk - please backup your stuff, this is master based.

Use:
```
<advancedsettings>
  <gui>
   <transparentguilayout>true</transparentguilayout>
 </gui>
</advancedsettings>
```

for testing. I am currently travelling (public holidays in Germany) I did not runtime test / debug it ... it might crash on start right away.

**Comment by Kristijan1001** [comment]
> First testbuild: https://jenkins.kodi.tv/job/android-arm-docker/6548/artifact/kodi-20240509-62ad1ec2-decorview-armeabi-v7a.apk - please backup your stuff, this is master based.
> 
> Use:
> 
> ```
> <advancedsettings>
>   <gui>
>    <transparentguilayout>true</transparentguilayout>
>  </gui>
> </advancedsettings>
> ```
> 
> for testing. I am currently travelling (public holidays in Germany) I did not runtime test / debug it ... it might crash on start right away.

OK Tested. Gui Stuf
![20240510_003937](https://github.com/xbmc/xbmc/assets/24441639/7b6d2719-b8a4-45c0-b984-09c7247d06f2)
f for HDR doesnt change the Gray Bars.

The build you provided crashes Kodi on startup, the start up bar filles up then crashes. Then crashes instantly on every boot.

Here is picture of how it looks:

**Comment by fritsch** [comment]
Debuglog would be great from the testbuild. But yeah - I expected that ;-) but luck would sometimes be great. I will change the position when I change to transparency.

**Comment by fritsch** [comment]
Good: https://jenkins.kodi.tv/job/android-arm-docker/6556/artifact/kodi-20240510-7f87bcec-decorview-armeabi-v7a.apk - runs for me.

**Comment by Kristijan1001** [comment]
> New test: Good: https://jenkins.kodi.tv/job/android-arm-docker/6556/artifact/kodi-20240510-7f87bcec-decorview-armeabi-v7a.apk - runs for me.

No changes, Unless the a
[advancedsettings.txt](https://github.com/xbmc/xbmc/files/15276791/advancedsettings.txt)
dvanced settings xml structure is wrong, Here it is just to check and make sure:

**Comment by fritsch** [comment]
Post your debuglog please.

**Comment by Kristijan1001** [comment]
> Post your debuglog please.

https://paste.kodi.tv/voposobomu

**Comment by fritsch** [comment]
2024-05-11 09:19:27.508 T:10471    info <general>: No settings file to load (special://xbmc/system/advancedsettings.xml)
2024-05-11 09:19:27.509 T:10471    info <general>: No settings file to load (special://masterprofile/advancedsettings.xml)

**Comment by Kristijan1001** [comment]
> 2024-05-11 09:19:27.508 T:10471 info : No settings file to load (special://xbmc/system/advancedsettings.xml) 2024-05-11 09:19:27.509 T:10471 info : No settings file to load (special://masterprofile/advancedsettings.xml)

Sorry rookie mistake had it as .txt. still gray tho here is log

https://paste.kodi.tv/opozefogom

**Comment by fritsch** [comment]
Yeah - here - I am still blindly building. As Android start is partly asynchroneous I want to find the sweet spot to trigger it without running into issues: https://jenkins.kodi.tv/job/android-arm-docker/6595/ <- please try the resulting APk once it is finished - it will print something.

It might crash again - due to setting the value on display change ... let's find out. Thanks.

**Comment by Kristijan1001** [comment]
> Yeah - here - I am still blindly building. As Android start is partly asynchroneous I want to find the sweet spot to trigger it without running into issues: https://jenkins.kodi.tv/job/android-arm-docker/6595/ <- please try the resulting APk once it is finished - it will print something.
> 
> It might crash again - due to setting the value on display change ... let's find out. Thanks.
> 
> https://jenkins.kodi.tv/job/android-arm-docker/6595/artifact/kodi-20240511-65db38bc-decorview-armeabi-v7a.apk <- here
> 
> PS: Starting next week I have my environment back - and won't send blind changes ;-)

Crashes on Startup

**Comment by fritsch** [comment]
logfile please :-)  - but I create you an update.

**Comment by fritsch** [comment]
And the weekend's final: https://jenkins.kodi.tv/job/android-arm-docker/6619/artifact/kodi-20240512-b23c7186-decortest-armeabi-v7a.apk

**Comment by Kristijan1001** [comment]
> And the weekend's final: https://jenkins.kodi.tv/job/android-arm-docker/6619/artifact/kodi-20240512-b23c7186-decortest-armeabi-v7a.apk

Sorry for late reply. Gray bars still. Thanks.

https://paste.kodi.tv/uquqaqequx

**Comment by fritsch** [comment]
Okay - too bad - code is called. Let's wait for the others input in the Pull-Request.

**Comment by quietvoid** [comment]
I'm still able to reproduce the #22561 issue on my device with Kodi 20.5 Nexus.
It doesn't happen with the build from #25171, so the background is set differently it seems.

I might try to figure it out if I have time.

**Comment by fritsch** [comment]
So - you meean - the PR works for you? I am personally not yet happy, cause of yet another ifdef in application - but - it's not so easy to find the right spot, right after gui and splash is initialized ...

**Comment by quietvoid** [comment]
> So - you meean - the PR works for you
No, it doesn't work.

**Comment by Kristijan1001** [comment]
So this cannot be fixed?

**Comment by fritsch** [comment]
Did Sony answer this to your support request? Please do so and post their official reply here, then it's at least documented and we have a public official statement to see if they "can/ want" fix it or not ...

Kodi cannot fix it in general, it can only workaround / mitigate it for Sony ... work week here -> busy.

**Comment by Kristijan1001** [comment]
> Did Sony answer this to your support request? Please do so and post their official reply here, then it's at least documented and we have a public official statement to see if they "can/ want" fix it or not ...
> 
> Kodi cannot fix it in general, it can only workaround / mitigate it for Sony ... work week here -> busy.

I never posted a support request to Sony, This is not an issue that also appeared recently. It has been like this for years. There have been many people that have asked about this and it has never been fixed. Me asking is just another needle in the stack. If they wanted to or could fix it, they would have already. This is a 5 Year old TV. Even if they answered, you would get the Typical PR answer, they will never say if they can fix it or want to fix it. Anyways i know it is a workaround i really hope this gets implemented. Thanks for your work so far it is appreciated.

**Comment by Kristijan1001** [comment]
> Hello folks! I also own a Sony TV. Model: SONY KD-55XE9005
> 
> Since the update via Google Play Store, the following problems have occurred:
> 
>     1. Kodi start/boot image destroyed, start image barely recognizable, shifted, blurred, - in any case, the boot image is faulty!
> 
>     2. Films where the upper and lower black bars have been removed now have gray bars! These are even displayed a little brighter in darker film images/frames and a little darker in dark scenes. - Looks terrible, you can no longer concentrate on the film!
>        This applies to all films that have HDR10, HDR10+ and Dolby Vision content! - In films where black bars have not been removed from the video, the bars remain black.
> 
> 
> I like Kodi, and I am also grateful for the program and have supported the project financially several times, but since the last update I have been very, very unhappy! Especially because of the gray bars that were previously black in HDR videos! (There are many people who also have these problems now, but don't bother to post them in the forum)
> 
> Question: How can I undo the update from version 21 without having to re-enter all of my painstakingly entered network links?
> 
> Best regards from Germany

Yeah it is a weird decision from the Kodi Team to break support for one devices, just to add support for other devices. The trade off just doesnt make sense, all HDR Content broken on pletora of devices, just to fix a bug with Dolby Vision on some devices which is not really widly used much, not even a windows desktop pc supports dolby vision. Kodi is now unusable. Seems really niche to break normal generic playback for Sony Android TVs , just to add niche support for dolby where not many people will even use it, and even if it didnt work it can always fall back to HDR instead. So we all have at least usable Kodi. Honestly i lost hope for this and went back to the old Kodi build. I will use it until addons no longer work i guess, but in general this sucks.

**Comment by neo1973** [comment]
> Kodi start/boot image destroyed, start image barely recognizable, shifted, blurred, - in any case, the boot image is faulty!

That's the intended look 😉 (see https://github.com/xbmc/xbmc/blob/21.0-Omega/media/splash.jpg)

**Comment by Ragnaroeko** [comment]
> > Hello folks! I also own a Sony TV. Model: SONY KD-55XE9005
> > Since the update via Google Play Store, the following problems have occurred:
> > ```
> > 1. Kodi start/boot image destroyed, start image barely recognizable, shifted, blurred, - in any case, the boot image is faulty!
> > 
> > 2. Films where the upper and lower black bars have been removed now have gray bars! These are even displayed a little brighter in darker film images/frames and a little darker in dark scenes. - Looks terrible, you can no longer concentrate on the film!
> >    This applies to all films that have HDR10, HDR10+ and Dolby Vision content! - In films where black bars have not been removed from the video, the bars remain black.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I like Kodi, and I am also grateful for the program and have supported the project financially several times, but since the last update I have been very, very unhappy! Especially because of the gray bars that were previously black in HDR videos! (There are many people who also have these problems now, but don't bother to post them in the forum)
> > Question: How can I undo the update from version 21 without having to re-enter all of my painstakingly entered network links?
> > Best regards from Germany
> 
> Yeah it is a weird decision from the Kodi Team to break support for one devices, just to add support for other devices. The trade off just doesnt make sense, all HDR Content broken on pletora of devices, just to fix a bug with Dolby Vision on some devices which is not really widly used much, not even a windows desktop pc supports dolby vision. Kodi is now unusable. Seems really niche to break normal generic playback for Sony Android TVs , just to add niche support for dolby where not many people will even use it, and even if it didnt work it can always fall back to HDR instead, then we will all have at least usable Kodi. Honestly i lost hope for this and went back to the old Kodi build. I will use it until addons no longer work i guess, but in general this sucks.

As far as I know, my SONY KD-55XE9005 doesn't know Dolby Vision at all, it can only do RGB, HLG and HDR10 formats, and I don't have much else to choose from in the Sony settings menu. I changed the color values ​​on the Sony TV from BT 2020 to BT 709 and also tested RGB. Afterwards, there were no useful changes when playing with Kodi. The films where the black bars were removed still showed gray bars.

As already written, films where the black bars were not removed still have their black bars, unfortunately most of my films have removed black bars, so I now have these ugly gray bars in many films. I converted my entire Blu-ray collection that I own with Staxrip, and removed the black bars with Crop to save storage space and made the video image a bit larger.

I don't think Sony will ever be interested in fixing this error or problem. Because original films with black bars don't cause any problems! It only affects the converted films where the black bars have been removed! Sony could see something like that as a pirated copy and will therefore hardly change anything about it 😉 since I have a lot of original films, I wanted to copy a large part of them to my hard drive and then stream them via Kodi on my Android SONY KD-55XE9005, so I don't have to constantly search for or change a Blu-ray, it's much more convenient.

It worked wonderfully until I installed the new Kodi update. I bought an Android TV years ago specifically so I didn't have to run everything via an external Android box, even if my Sony can only do 1080p via Android, the Android hardware in the Sony doesn't do more than that. 4K only works via HDMI, which isn't a problem, the film picture in 1080p is good enough.

As I wrote, it's not necessarily a mistake by Sony, but perhaps more of a measure against non-original Blu-ray disc material where the black bars have been removed. A friend of mine has a new Android Sony TV, and he has the same problem as me since the Kodi update and his TV is NEW and up to date!

But that doesn't help us now, I won't buy another brand or a new TV as long as mine is still good (besides, my TV wasn't exactly cheap or inexpensive years ago). Maybe the Kodi program authors will have mercy and build something into the Kodi software that can help people like us Sony TV owners to get black bars in the videos again. That would be nice, thanks!!!

Best regards

**Comment by Ragnaroeko** [comment]
> > Kodi start/boot image destroyed, start image barely recognizable, shifted, blurred, - in any case, the boot image is faulty!
> 
> That's the intended look 😉 (see https://github.com/xbmc/xbmc/blob/21.0-Omega/media/splash.jpg)

Yes, exactly, that's what the Kodi boot logo looks like on my SONY KD-55XE9005, a bit destroyed since the update!

Best regards

**Comment by Ragnaroeko** [comment]
Forgot, here again all the errors I have had since updating to Kodi 21:

1. Kodi boot/start logo destroyed

2. Films where the bars were removed now have gray transparent bars instead of black bars.

3. Subtitle texts are now gray instead of white!

I forgot to mention point 3!

Best regards

**Comment by neo1973** [comment]
> Kodi boot/start logo destroyed

It's not destroyed, that is the intended look. It's an artistic choice.

**Comment by Ragnaroeko** [comment]
> > Kodi boot/start logo destroyed
> 
> It's not destroyed, that is the intended look. It's an artistic choice.

aha... Personally, I don't think that was the best decision 😉

Best regards

Thanks for the info!

**Comment by fritsch** [comment]
For Sony, which does not care for their buying users at all: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk

**Comment by Kristijan1001** [comment]
> For Sony, which does not care for their buying users at all: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk

It works Thank you very much really apricate it. Note to SONY users, if ur TV doesnt have HDR+ dont disable it only disable Dolby Other wise with Dolby Enabled movies will lag and with both disabled they will also lag again. Turning off adjust HDR also lags if i remember correctly.  Contrarily to the Retail Version where disabling both HDR10+ and Dolby wasnt lagging.

**Comment by Ragnaroeko** [comment]
> Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk

Link ist not working hmmmmmm,.....

Can you help me please? THANX from germany

**Comment by Kristijan1001** [comment]
> > Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk
> 
> Link ist not working hmmmmmm,.....
> 
> Can you help me please? THANX from germany

https://www.mediafire.com/file/v9ubxfqiduptbnn/kodi-20240529-e25b9b88-sony-armeabi-v7a+(1).apk/file

Here you go, dunno if it gets auto deleted but this will prolly go in the main kodi build at some point i assume ?

**Comment by Ragnaroeko** [comment]
> > > Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk
> > 
> > 
> > Der Link funktioniert nicht, hmmmmmm,.....
> > Könnt ihr mir bitte helfen? DANKE aus Deutschland
> 
> https://www.mediafire.com/file/v9ubxfqiduptbnn/kodi-20240529-e25b9b88-sony-armeabi-v7a+(1).apk/file
> 
> Hier bitte, ich weiß nicht, ob es automatisch gelöscht wird, aber ich nehme an, dass es wahrscheinlich irgendwann in die Hauptversion von Kodi aufgenommen wird?


Thank you, thank you! :-) The main thing is that I can watch my videos normally again.

Does anyone know if you can just install it or will my network settings be lost?

Greetings

**Comment by Kristijan1001** [comment]
> > > > Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk
> > > 
> > > 
> > > Der Link funktioniert nicht, hmmmmmm,.....
> > > Könnt ihr mir bitte helfen? DANKE aus Deutschland
> > 
> > 
> > https://www.mediafire.com/file/v9ubxfqiduptbnn/kodi-20240529-e25b9b88-sony-armeabi-v7a+(1).apk/file
> > Hier bitte, ich weiß nicht, ob es automatisch gelöscht wird, aber ich nehme an, dass es wahrscheinlich irgendwann in die Hauptversion von Kodi aufgenommen wird?
> 
> Thank you, thank you! :-) The main thing is that I can watch my videos normally again.
> 
> Does anyone know if you can just install it or will my network settings be lost?
> 
> Greetings

i would suggest making a backup of kodi, this is still experimental and u dont want to screw up ur main build for the future

**Comment by Ragnaroeko** [comment]
> > > > > Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk
> > > > 
> > > > 
> > > > Der Link funktioniert nicht, hmmmmmm,.....
> > > > Könnt ihr mir bitte helfen? DANKE aus Deutschland
> > > 
> > > 
> > > https://www.mediafire.com/file/v9ubxfqiduptbnn/kodi-20240529-e25b9b88-sony-armeabi-v7a+(1).apk/file
> > > Hier bitte, ich weiß nicht, ob es automatisch gelöscht wird, aber ich nehme an, dass es wahrscheinlich irgendwann in die Hauptversion von Kodi aufgenommen wird?
> > 
> > 
> > Thank you, thank you! :-) The main thing is that I can watch my videos normally again.
> > Does anyone know if you can just install it or will my network settings be lost?
> > Greetings
> 
> i would suggest making a backup of kodi usedata, this is still experimental and u dont want to screw up ur main build for the future

If only I hadn't updated to the new version, everything worked before. Now I have to spend another 2 or 3 hours resetting all the settings I had... if it works, I don't think I'll update again as long as the TV is alive ;-)

**Comment by Kristijan1001** [comment]
> > > > > > Für Sony, das sich überhaupt nicht um seine kaufenden Benutzer kümmert: https://jenkins.kodi.tv/job/android-arm-docker/6741/artifact/kodi-20240529-e25b9b88-sony-armeabi-v7a.apk
> > > > > 
> > > > > 
> > > > > Der Link funktioniert nicht, hmmmmmm,.....
> > > > > Könnt ihr mir bitte helfen? DANKE aus Deutschland
> > > > 
> > > > 
> > > > https://www.mediafire.com/file/v9ubxfqiduptbnn/kodi-20240529-e25b9b88-sony-armeabi-v7a+(1).apk/file
> > > > Hier bitte, ich weiß nicht, ob es automatisch gelöscht wird, aber ich nehme an, dass es wahrscheinlich irgendwann in die Hauptversion von Kodi aufgenommen wird?
> > > 
> > > 
> > > Thank you, thank you! :-) The main thing is that I can watch my videos normally again.
> > > Does anyone know if you can just install it or will my network settings be lost?
> > > Greetings
> > 
> > 
> > i would suggest making a backup of kodi usedata, this is still experimental and u dont want to screw up ur main build for the future
> 
> If only I hadn't updated to the new version, everything worked before. Now I have to spend another 2 or 3 hours resetting all the settings I had... if it works, I don't think I'll update again as long as the TV is alive ;-)

You can still use your old settings, just make a backup of them.

**Comment by quietvoid** [comment]
@Kristijan1001 @Ragnaroeko Here's a build from #25300: https://jenkins.kodi.tv/job/android-arm-docker/6807/artifact/kodi-20240602-8078f224-PR25300-armeabi-v7a.apk
Can someone test it?

**Comment by Kristijan1001** [comment]
> @Kristijan1001 @Ragnaroeko Here's a build from #25300: https://jenkins.kodi.tv/job/android-arm-docker/6807/artifact/kodi-20240602-8078f224-PR25300-armeabi-v7a.apk
> 
> Can someone test it? Advanced setting needs to be enabled:
> 
> ```
> <advancedsettings version="1.0">
>   <gui>
>     <transparentvideolayout>true</transparentvideolayout>
>   </gui>
> </advancedsettings>
> ```

Sorry i have been busy should be able to test it tmrw probably. Can you relink the build because the link is dead. Is there anything diffrent about this because the previous build worked just fine.

**Comment by quietvoid** [comment]
@Kristijan1001 You can find a build of #25300 here: https://github.com/quietvoid/xbmc/releases/download/sony-25300/kodi-21-Omega-armeabi-v7a-release-quietvoid-sony-2024-06-09.apk  
It installs separately from regular Kodi, package name is `org.xbmc.kodi.quietvoid`

> Is there anything diffrent about this because the previous build worked just fine.

I don't know what @fritsch changed but I think it just reverted to the old Kodi 20 Sony workaround.

Now there's an advanced setting to apply the old workaround.
```
<advancedsettings version="1.0">
  <gui>
    <transparentvideolayout>true</transparentvideolayout>
  </gui>
</advancedsettings>
```

**Comment by Kristijan1001** [comment]
Sorry for posting in closed issue, when is this getting merged into the main kodi build would it be nightly or release? Thank you.

**Comment by quietvoid** [comment]
It should already be in the recent nightlies.
And it would be in the next Kodi 21 Omega release

**Comment by Kristijan1001** [comment]
> @Kristijan1001 You can find a build of #25300 here: https://github.com/quietvoid/xbmc/releases/download/sony-25300/kodi-21-Omega-armeabi-v7a-release-quietvoid-sony-2024-06-09.apk It installs separately from regular Kodi, package name is `org.xbmc.kodi.quietvoid`
> 
> > Is there anything diffrent about this because the previous build worked just fine.
> 
> I don't know what @fritsch changed but I think it just reverted to the old Kodi 20 Sony workaround.
> 
> Now there's an advanced setting to apply the workaround instead, so that Sony TV users can fix it without it impacting others.
> 
> ```
> <advancedsettings version="1.0">
>   <gui>
>     <transparentvideolayout>true</transparentvideolayout>
>   </gui>
> </advancedsettings>
> ```

Hello, i just updated to kodi 21.1 and this broke. What happened ? Is there a new setting i need to do?

**Comment by quietvoid** [comment]
There's no new setting, it's supposed to still be there in 21.1.
You can always check the logs for the line `XBMCApp: VideoLayout view was set to transparent.`

**Comment by Kristijan1001** [comment]
> There's no new setting, it's supposed to still be there in 21.1. You can always check the logs for the line `XBMCApp: VideoLayout view was set to transparent.`

https://paste.kodi.tv/erevazefup

I think it does get called i changed nothing just updated from google play from the version u uploaded long ago.

**Comment by quietvoid** [comment]
The reason is that the setting has changed name since the build, you're still using `transparentguilayout`.
It's `transparentvideolayout` now.

`<transparentvideolayout>true</transparentvideolayout>`

**Comment by Kristijan1001** [comment]
> The reason is that the setting has changed name since the build, you're still using `transparentguilayout`. It's `transparentvideolayout` now.
> 
> `<transparentvideolayout>true</transparentvideolayout>`

It works thanks 👍
